src/tracing/esp32_trace/include/matter/tracing/macros_impl.cpp

```diff
@@ -17,8 +17,135 @@
 #include "macros_impl.h"
 #include <esp_heap_caps.h>
 #include <esp_insights.h>
+
 namespace Insights {
 
+/* The traces of PacketParser,MinMdnsResolver and SesssionManager are filtered out :
+ * i) To report the traces of more concern during commissioning like CASESession,PASESession,
+ * traces for OperationalSessionSetup, GeneralCommissioning and NetworkCommissioning
+ * ii) although providing the user with ablity to add or remove the strings whose traces
+ * can be filtered out.
+ */
+namespace {
+static constexpr size_t kBlackListSize = 10;
+static bool isBlacklistInitialized     = false;
+static size_t blacklistHashes[kBlackListSize];
+static const char * blackListStrings[kBlackListSize] = {
+    "PacketParser",
+    "MinMdnsResolver",
+    "SessionManager",
+    "DeviceCommissioner",
+};
+
+static size_t MurmurHash(const void * key)
+{
+    const size_t multiplier    = 0x5bd1e995;
+    const size_t shift         = 24;
+    const unsigned char * data = (const unsigned char *) key;
+    size_t hash                = 0;
+
+    while (*data)
+    {
+        size_t value = *data++;
+        value *= multiplier;
+        value ^= value >> shift;
+        value *= multiplier;
+        hash *= multiplier;
+        hash ^= value;
+    }
+
+    hash ^= hash >> 13;
+    hash *= multiplier;
+    hash ^= hash >> 15;
+
+    return hash;
+}
+
+static bool IsBlacklistedString(const char * input)
+{
+    size_t hashValue = MurmurHash(input); // Seed is always zero
```

Comment:
There is no seed here in the call. Maybe this should be in a comment of the hash function (like implements murmurhash with 0 seed)